pkg/watcher/metricsprovider.go

```diff
@@ -50,11 +50,14 @@ type MetricsProviderClient interface {
 	FetchHostMetrics(host string, window *Window) ([]Metric, error)
 	// Fetch metrics for all hosts
 	FetchAllHostsMetrics(window *Window) (map[string][]Metric, error)
+	// Get metric provider server health status
+	// Returns 0 if healthy, -1 if unhealthy along with error if any
+	Health() (int, error)
```

Comment:
if it is 0 or 1? why don't we just get rid of it?
error can bit nil or not nil?

Reply:
error will help to customize message internally. also they may not be present always if unhealthy . Status codes will be used for metrics in the future.